### PR TITLE
BPR-120: Refactor C# architecture

### PR DESCRIPTION
**Pull request by vladdirlea2002**

### CODE REVIEW

File: BPR.Analysis/Services/AnalysisService.cs [hunk 1]
```diff
@@ -13,64 +18,77 @@ public AnalysisService(ICodeExtractionService codeExtractionService)
...
 
         return violations;
     }
 
-    private async Task<List<Violation>> GetDependencyAnalysisAsync(string folderPath, AnalysisArchitecturalModel model)
+    private async Task<List<Violation>> GetDependencyAnalysisAsync(string folderPath, ArchitecturalModel model)
     {
         var projectNames = _codeExtractionService.GetProjectNames(folderPath);
-        List<Violation> violations = await GetDependencyAnalysisOnNamespaceComponentsAsync(folderPath, projectNames, model.Components);
+        List<Violation> violations =
```
Comment (stefaniatomuta) [inline BPR.Analysis/Services/AnalysisService.cs:42]: Use var

File: BPR.Analysis/Services/AnalysisService.cs [hunk 2]
```diff
@@ -128,4 +148,65 @@ internal static List<Violation> GetNamespaceAnalysis(List<NamespaceDirective> na
...
+
+    private async Task<List<NamespaceDirective>> GetNamespaceDirectivesAsync(string folderPath)
+    {
+        List<NamespaceDirective> matches = new();
+        var files = Directory.GetFiles(folderPath, "*", SearchOption.AllDirectories);
+        foreach (var file in files)
+        {
+            if (file.EndsWith(".cs") || file.EndsWith(".cshtml"))
+            {
+                var content = await File.ReadAllLinesAsync(file, Encoding.UTF8);
```
Comment (stefaniatomuta) [inline BPR.Analysis/Services/AnalysisService.cs:189]: I see some similar code with GetUsingDirectivesAsync, maybe a method can help to diminish code duplication

File: BPR.Analysis/Services/CodeExtractionService.cs [hunk 3]
```diff
@@ -42,7 +13,7 @@ public List<string> GetProjectNames(string folderPath)
         foreach (var dir in projDirectories)
         {
             files.AddRange(Directory.GetFiles(dir)
-                 .Where(file => file.EndsWith("csproj")));
+                .Where(file => file.EndsWith("csproj")));
```
Comment (stefaniatomuta) [inline BPR.Analysis/Services/CodeExtractionService.cs:16]: What about an Enum for this and cshtml as they are used in many places?

File: BPR.Analysis/Services/AnalysisService.cs (hunk 2 above)
Comment (vladdirlea2002) [reply BPR.Analysis/Services/AnalysisService.cs:189]: Can't see an easy way to do so. better in another PR as I touched way too many here

**Review by stefaniatomuta — APPROVED** [review]

**Pull request merged**